Make FFI public?

- Working on #1128

I've gotten [this far](https://github.com/opendp/opendp-rust-example/pull/1/commits/0a81fee) and seem to be blocked because ffi is not public:

```
error[E0603]: module `ffi` is private
   --> src/lib.rs:11:13
    |
11  | use opendp::ffi::any::AnyDomain;
    |             ^^^  --- module `any` is not publicly re-exported
    |             |
    |             private module
    |
note: the module `ffi` is defined here
   --> /Users/chuck/.cargo/registry/src/index.crates.io-6f17d22bba15001f/opendp-0.8.0/src/lib.rs:170:1
    |
170 | mod ffi;
    | ^^^^^^^
```

Michael notes:
> seems like we might want that to be public, so people can write their own FFI
>
> There are functions you can use to access the AnyDomain inside of an AnyTransformation or AnyMeasurement. But I guess you can't make an instance of one yourself unless it's made public (edited)

--- a/rust/src/ffi/mod.rs
+++ b/rust/src/ffi/mod.rs
@@ -3,6 +3,8 @@
 //! # Overview
 //!
 //! This module contains utilities necessary for the FFI bindings.
+//! It is public so you can write your own lightweight FFI for quick one-off language integrations,
+//! but its use should be rare: More often, you should use the Rust API directly.
 //!
 //! # Generic Functions
 //!

--- a/rust/src/lib.rs
+++ b/rust/src/lib.rs
@@ -167,7 +167,7 @@ macro_rules! enclose {
 
 #[cfg(feature = "ffi")]
 #[macro_use]
-mod ffi;
+pub mod ffi;
 #[cfg(feature = "ffi")]
 #[macro_use]
 extern crate lazy_static;

